bitcoin/src/taproot.rs

```diff
@@ -743,9 +757,17 @@ pub struct ControlBlock {
 }
 
 impl ControlBlock {
-    /// Constructs a `ControlBlock` from slice. This is an extra witness element that provides the
-    /// proof that taproot script pubkey is correctly computed with some specified leaf hash. This
-    /// is the last element in taproot witness when spending a output via script path.
+    /// Constructs a `ControlBlock` from slice.
+    #[deprecated(since = "0.30.0", note = "Use decode instead")]
+    pub fn from_slice(sl: &[u8]) -> Result<ControlBlock, TaprootError> {
+        Self::decode(sl)
+    }
+
+    /// Constructs a `ControlBlock` from slice.
```

Comment:
This performs decoding from witness element, so it should say decode or something. Similar to `TaprootMerkleBranch`.

Reply:
I went with:
```
    /// Decodes bytes representing a `ControlBlock`.
```